dotcom-rendering/src/web/components/MainMedia.tsx

```diff
@@ -70,29 +70,34 @@ export const MainMedia: React.FC<{
 	host,
 	pageId,
 	webTitle,
-}) => (
-	<div
-		css={[
-			mainMedia,
-			format.display === ArticleDisplay.Immersive
-				? immersiveWrapper
-				: noGutters,
-		]}
-	>
-		{elements.map((element, index) =>
-			renderArticleElement({
-				format,
-				palette,
-				element,
-				adTargeting,
-				host,
-				index,
-				isMainMedia: true,
-				starRating,
-				hideCaption,
-				pageId,
-				webTitle,
-			}),
-		)}
-	</div>
-);
+}) => {
+	const isImmersive = format.display === ArticleDisplay.Immersive;
+	const isLiveOrDeadBlog =
+		format.design === ArticleDesign.LiveBlog ||
+		format.design === ArticleDesign.DeadBlog;
+	return (
+		<div
+			css={[
+				mainMedia,
+				isImmersive && immersiveWrapper,
+				!isImmersive && !isLiveOrDeadBlog && noGutters,
+			]}
+		>
```

Comment:
This code looks like its reached the point where it could benefit from a refactor to make it clearer?